components/brave_vpn/common/wireguard/wireguard_utils.cc

```diff
@@ -82,6 +88,102 @@ WireguardKeyPair GenerateNewX25519Keypair() {
       EncodeBase64(std::vector<uint8_t>(privkey, privkey + 32)));
 }
 
+bool ValidateKey(const wchar_t* key,
+                 std::string* output,
+                 const std::string field_name) {
+  if (!base::WideToUTF8(key, wcslen(key), output)) {
+    VLOG(1) << "failed decoding " << field_name;
+    return false;
+  }
+
+  base::TrimWhitespaceASCII(*output, base::TRIM_ALL, output);
```

Comment:
so backtracking a bit on the comments that were made in a previous version about using chromium utilities, given that this is untrusted input, I think the first thing we need to do is pass it through re2 assuming that @thypon agrees that the regex results in safe output.

```
As another special case, we trust the [RE2](https://cs.chromium.org/chromium/src/third_party/re2/README.chromium) regular expression library to evaluate untrustworthy patterns over untrustworthy input strings, because its grammar is sufficiently limited and hostile input is part of the threat model against which it's been tested for years. It is not the case, however, that text matched by an RE2 regular expression is necessarily “sanitized” or “safe”. That requires additional security judgment.
```

Reply:
We are gonna remove `TrimWhitespaceASCII` altogether

Reply:
Removed.